tests/integration/test_coverage_gap_supplements.py

```
@@ -94,6 +93,11 @@ def _make_embedder_with_fake_sklearn(**kw: Any) -> DocumentEmbedder:
 class TestDocumentEmbedderWithFakeSklearn:
     """Cover lines 56-322 in embedder.py via a fake TfidfVectorizer."""
 
+    @pytest.fixture(autouse=True)
+    def _require_numpy(self) -> None:
+        global np
```

Ruff (0.15.10)

[warning] Using the global statement to update `np` is discouraged

(PLW0603)